src/Windows/Avalonia.Direct2D1/Media/DrawingContextImpl.cs

```diff
@@ -100,16 +102,43 @@ public void Dispose()
         /// <param name="opacity">The opacity to draw with.</param>
         /// <param name="sourceRect">The rect in the image to draw.</param>
         /// <param name="destRect">The rect in the output to draw to.</param>
-        public void DrawImage(IRef<IBitmapImpl> source, double opacity, Rect sourceRect, Rect destRect)
+        /// <param name="scalingMode"></param>
+        public void DrawImage(IRef<IBitmapImpl> source, double opacity, Rect sourceRect, Rect destRect, BitmapScalingMode scalingMode)
         {
             using (var d2d = ((BitmapImpl)source.Item).GetDirect2DBitmap(_renderTarget))
             {
-                _renderTarget.DrawBitmap(
-                    d2d.Value,
-                    destRect.ToSharpDX(),
-                    (float)opacity,
-                    BitmapInterpolationMode.Linear,
-                    sourceRect.ToSharpDX());
+                if (_renderTarget is DeviceContext deviceContext)
+                {
+                    deviceContext.DrawBitmap(d2d.Value, destRect.ToDirect2D(), (float)opacity, GetInterpolationMode(scalingMode), sourceRect.ToDirect2D(), null);
+                }
+                else
+                {
+                    var interpolationMode = scalingMode == BitmapScalingMode.LowQuality
```

Comment:
It means the MediumQuality same as HighQuality for it uses BitmapInterpolationMode.Linear

Reply:
I know what that means. This is just a limitation of D2D1.0 it will change after we use DeviceContext everywhere. D2D1.1 support better interpolation modes.